app/Main.hs

```diff
@@ -235,53 +241,61 @@ requestSecret opts secret =
             $ defaultRequest
 
     shouldRetry = const $ return . isLeft
-    retryAction _retryStatus = doRequest secret request
+    retryAction _retryStatus = doRequest secretPath request
   in
     Retry.retrying vaultRetryPolicy shouldRetry retryAction
 
-
+-- | Requests all the supplied secrets from the vault, but just once,
+-- even if multiple keys are specified for a single secret.
```

Comment:
Same as above.

Reply:
Please also explain that this is an optimization to avoid unnecessary round-trips (and DNS requests).